interfaces/cython/cantera/solutionbase.pyx

```diff
@@ -537,6 +543,29 @@ cdef class SolutionArrayBase:
             cxx_shape.push_back(dim)
         self.base.setApiShape(cxx_shape)
 
+    def info(self, keys=None, rows=10, width=80):
+        """
+        Print a concise summary of a `SolutionArray`.
+
+        :param keys: List of components to be displayed; if `None`, all components are
+            considered.
+        :param rows: Maximum number of rendered rows.
+        :param width: Maximum width of rendered output.
+        """
+        cdef vector[string] cxx_keys
+        if keys is not None:
+            for key in keys:
+                cxx_keys.push_back(stringify(key))
+        elif self._phase.selected_species:
+            keep = self._phase.species_names
+            self._phase.selected_species = []
+            names = self._phase.species_names
+            self._phase.selected_species = keep
+            for key in self.component_names:
+                if key not in names or key in keep:
+                    cxx_keys.push_back(stringify(key))
```

Comment:
Converting `names` and `keep` to sets before this loop would avoid making this an $O(n^2)$ operation.